src/base/Cellar.sol

```diff
@@ -1069,6 +1051,103 @@ contract Cellar is ERC4626, Ownable, Multicall {
 
     // ========================================= FEES LOGIC =========================================
 
+    /**
+     * @notice Stores the share price to be used as a High Watermark to calculate performance fees
+     */
+    uint256 public sharePriceHighWatermark;
+
+    /**
+     * @notice Number 0 -> 1e18 defining how much of the performance fee goes to the strategist
+     */
+    uint256 public strategistPerformanceCut = 0.75e18;
+
+    /**
+     * @notice Number 0 -> 1e18 defining how much of the platform fee goes to the strategist
+     */
+    uint256 public strategistPlatformCut = 0.75e18;
+
+    /**
+     * @notice Address to send the strategists fee shares
+     */
+    address public strategistPayoutAddress;
+
+    ///@dev resets high watermark to current share price
+    function resetHighWatermark() external onlyOwner {
+        sharePriceHighWatermark = totalAssets().mulDivDown(10**decimals, totalSupply);
+    }
+
+    ///@dev Sets the Strategists cut of performance fees
+    function setStrategistPerformanceCut(uint256 cut) external onlyOwner {
+        require(cut <= 1e18, "Invalid cut");
+        strategistPerformanceCut = cut;
+    }
+
+    ///@dev Sets the Strategists cut of platform fees
+    function setStrategistPlatformCut(uint256 cut) external onlyOwner {
+        require(cut <= 1e18, "Invalid cut");
+        strategistPlatformCut = cut;
+    }
+
+    ///@dev Sets the Strategists payout address
+    function setStrategistPayoutAddress(address payout) external onlyOwner {
+        strategistPayoutAddress = payout;
+    }
+
+    /**
+     * @notice Calculates how many assets Strategist would earn performance fees
+     * @param _totalAssets uint256 value of the total assets in the cellar
+     * @return feeInAssets amount of assets
+     */
+    function _calculatePerformanceFee(uint256 _totalAssets) internal view returns (uint256 feeInAssets) {
+        if (performanceFee == 0 || _totalAssets == 0) return 0;
+        uint256 currentSharePrice = _convertToAssets(10**decimals, _totalAssets);
+        if (sharePriceHighWatermark == 0) return 0;
+        else if (sharePriceHighWatermark < currentSharePrice) {
+            //find how many assets make up the fee
+            uint256 yield = ((currentSharePrice - sharePriceHighWatermark) * totalSupply) / 10**decimals;
+            feeInAssets = yield.mulDivUp(performanceFee, 1e18);
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * @notice Mints cellar performance fee shares if current share price is above high watermark
+     * @param _totalAssets uint256 value of the total assets in the cellar
+     */
+    function _takePerformanceFees(uint256 _totalAssets) internal {
+        if (performanceFee == 0) return;
```

Comment:
In order to 1) reduce the possible sources of error causing discrepencies between the amounts returned by deposit/withdraw functions and their `preview` counterparts and 2) reduce the amount of redundant code between `calculatePerformanceFee` and `_takePerformanceFees`, `_takePerformanceFees` and `_calculatePerformanceFees` should basically be the same except that `_takePerformanceFees` actually mints the fees while `_calculatePerformanceFees` just returns how much will be minted without actually minting them.

Also rename `_calculatePerformanceFees` to `_previewPerformanceFees`.